test/psych/test_safe_load.rb

```diff
@@ -22,7 +22,7 @@ class Foo; end
     def test_no_recursion
       x = []
       x << x
-      assert_raise(Psych::BadAlias) do
+      assert_raise(Psych::AliasesNotEnabled) do
         Psych.safe_load Psych.dump(x)
       end
     end
```

Comment:
We should fix this test, because I don't think it really makes sense:

Any alias will make this raise `Psych::AliasesNotEnabled`, because `alias: true` wasn't specified. The recursion is a red herring, any alias will trigger this exception to be thrown (and thus the assertion to pass).

Reply:
Yes, I think this is the correct fix.

Reply:
Opened https://github.com/ruby/psych/pull/568 to handle this